app/add-details/page.tsx

```diff
@@ -0,0 +1,134 @@
+'use client';
+
+import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+import { UUID } from 'crypto';
+import { updateUserPlants } from '@/api/supabase/queries/updateUserPlants';
+import PlantDetails from '@/components/PlantDetails';
+import { Plant, UserPlants } from '@/types/schema';
+
+export default function Home() {
+  const [currentIndex, setCurrentIndex] = useState<number>(1);
+  const [details, setDetails] = useState<Partial<UserPlants>[]>([]);
+  const router = useRouter();
+
+  const plants: Plant[] = [
+    {
+      id: '43c19f80-8205-4d03-b323-05c220550bf0',
+      plant_name: 'cabbbage',
+      us_state: 'string',
+      harvest_season: 'SPRING',
+      water_frequency: 'string',
+      weeding_frequency: 'string',
+      plant_seed_indoors_start: 'string',
+      plant_seed_indoors_end: 'string',
+      plant_seed_outdoors_start: 'string',
+      plant_seed_outdoors_end: 'string',
+      plant_transplant_start: 'string',
+      plant_transplant_end: 'string',
+      harvest_start: 'string',
+      harvest_end: 'string',
+      beginner_friendly: true,
+      plant_tips: 'string',
+      img: 'string',
+      difficulty_level: 'HARD',
+      sunlight_min_hours: 1,
+      sunlight_max_hours: 1,
+    },
+    {
+      id: '43c19f80-8205-4d03-b323-05c220550bf0',
+      plant_name: 'tomatoooooo',
+      us_state: 'string',
+      harvest_season: 'SPRING',
+      water_frequency: 'string',
+      weeding_frequency: 'string',
+      plant_seed_indoors_start: 'string',
+      plant_seed_indoors_end: 'string',
+      plant_seed_outdoors_start: 'string',
+      plant_seed_outdoors_end: 'string',
+      plant_transplant_start: 'string',
+      plant_transplant_end: 'string',
+      harvest_start: 'string',
+      harvest_end: 'string',
+      beginner_friendly: true,
+      plant_tips: 'string',
+      img: 'string',
+      difficulty_level: 'HARD',
+      sunlight_min_hours: 1,
+      sunlight_max_hours: 1,
+    },
+  ];
+  const user_id: UUID = 'e72af66d-7aae-45f6-935a-187197749d9f';
+
+  function move(steps: number) {
+    // if ur not at the end of the plant details flow update details to store what was in the inputs
+    if (currentIndex != plants.length + 1) {
+      const updatedDetails = [...details];
+      const plantID = plants[currentIndex - 1]['id'];
+      const date = (document.getElementById('date')! as HTMLInputElement).value;
```

Comment:
instead of doing `document.getElementById` we can use a useState to track the dateInput and plantingType, i.e. 
```
const [dateInput, setDateInput] = useState<Date | null>(null);
const [plantingType, setPlantingType] = useState<PlantingTypeEnum | null>(null);
```
(you might be able to handle the null cases differently, but this is just a suggestion) 
And then, pass the value and the setValue func (e.g. plantingType, setPlantingType) into `PlantDetails` (see the later comment)